--- a/src/dxvk/rtx_render/rtx_remix_api.cpp
+++ b/src/dxvk/rtx_render/rtx_remix_api.cpp
@@ -1019,8 +1019,8 @@
         dst.color0Buffer = dxvk::RasterBuffer { vertexSlice, offsetof(remixapi_HardcodedVertex, color), sizeof(remixapi_HardcodedVertex), VK_FORMAT_B8G8R8A8_UNORM };
         if (src.skinning_hasvalue) {
           dst.numBonesPerVertex = src.skinning_value.bonesPerVertex;
-          dst.blendWeightBuffer = dxvk::RasterBuffer { blendWeightsSlice, 0, sizeof(float), VK_FORMAT_R32_SFLOAT };;
-          dst.blendIndicesBuffer = dxvk::RasterBuffer { blendIndicesSlice, 0, sizeof(uint32_t), VK_FORMAT_R8G8B8A8_USCALED };
+          dst.blendWeightBuffer = dxvk::RasterBuffer { blendWeightsSlice, 0, sizeof(float) * src.skinning_value.bonesPerVertex, VK_FORMAT_R32_SFLOAT };
+          dst.blendIndicesBuffer = dxvk::RasterBuffer { blendIndicesSlice, 0, sizeof(uint32_t) * dxvk::divCeil(src.skinning_value.bonesPerVertex, 4u), VK_FORMAT_R8G8B8A8_USCALED };
         }
 
         dst.indexCount = src.indices_count;
